src/sinks/util/encoding.rs

```diff
@@ -156,7 +156,11 @@ pub trait EncodingConfiguration<E> {
                 Event::Log(log_event) => {
                     let to_remove = log_event
                         .keys()
-                        .filter(|f| !only_fields.contains(f))
+                        .filter(|f| {
+                            !only_fields
+                                .iter()
+                                .any(|only| f.starts_with(&(only.to_string() + "[")) || only == f)
```

Comment:
Can you think of any ways to break this easily?

Reply:
Can we use `PathIter` to get a `Vec<PathComponent>` and do prefix checking with that instead of strings?

Reply:
Oh yesss that would work much better.

Reply:
Ok that should be safer. :)